File: 1493-longest-subarray-of-1s-after-deleting-one-element/1493-longest-subarray-of-1s-after-deleting-one-element.py
class Solution:
    def longestSubarray(self, nums: List[int]) -> int:
        cnt0, longestWin, start = 0, 0, 0
        for i in range(0, len(nums)):
            if nums[i] == 0: cnt0 += 1
            while cnt0 > 1:
                if nums[start] == 0: cnt0 -= 1
                start += 1
            longestWin = max(longestWin, i-start)
        return longestWin

File: 1493-longest-subarray-of-1s-after-deleting-one-element/1493-longest-subarray-of-1s-after-deleting-one-element.java
class Solution {
    public int longestSubarray(int[] nums) {
        int cnt0 = 0, longestWin = 0, start = 0;
        for(int i=0; i<nums.length; i++){
            cnt0 += (nums[i]==0 ? 1 : 0);
            while(cnt0 > 1){
                cnt0 -= (nums[start]==0 ? 1 : 0);
                start++;
            }
            longestWin = Math.max(longestWin, i-start);
        }
        return longestWin;
    }
}